Reworked BakedAnimation and Frames to not load for every entity

opl- opened the pull request:
By @Fadest 

Use data loaded from animation files on disk on load by cloning it when spawning new baked entities. Old behavior was to reload and deserialize the data on baked entity spawn.

opl- commented on src/main/java/com/mcmiddleearth/entities/entities/composite/animation/BakedAnimationTree.java:
```diff
@@ -104,15 +135,22 @@ public BakedAnimation getAnimation(String[] path) {
-                //entity.getActionType().name().toLowerCase()
+            entity.getMovementType().name().toLowerCase(),
+            entity.getMovementSpeedAnimation().name().toLowerCase(),
+            getActionType(entity).name().toLowerCase(Locale.ROOT)
```
I think this was done on purpose and will break animation lookups if changed?

Fadest replied:
port/magma-1.16 has the same getActionType() method, that's the one that should be used, the feature/rework_animation_frames branch also has one but without the goals implemented in magma

opl- commented:
The merge conflict is caused by changes from https://github.com/CarperCreative/MCME-Entities/tree/fix/random-animations

opl- commented:
Cleaned up and rebased onto `development` due to conflicts with `fix/random-animations`. Original in `feature/rework_animations_frames-original`. Cherry picked into `port/magma-1.16` in b42cb0896037363a705845c5bb89b266ef514eb4.
